### ISSUE TITLE
Unable to open 'debug-eval': Unable to read file (Error: File not found

### ISSUE THREAD

**Issue by tiansin** [DEVSENSE/phptools-docs#38]
```
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/base/common/console.js:103 [Extension Host] Unhandled Rejection:  Error: socket not writable
    at write.Promise (/Users/tianxin/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugConnection.js:1:1579)
    at new Promise (<anonymous>)
    at Mt.write (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugConnection.js:1:1515)
    at Mt.<anonymous> (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:8544)
    at Generator.next (<anonymous>)
    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:375)
    at new Promise (<anonymous>)
    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:152)
    at Mt.F (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:8261)
    at Mt.G (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:8229)
    at B.Promise (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:8001)
    at new Promise (<anonymous>)
    at Mt.B (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:7976)
    at Mt.<anonymous> (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:10359)
    at Generator.next (<anonymous>)
    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:375)
    at new Promise (<anonymous>)
    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:152)
    at Mt.sendStopCommand (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:10301)
    at ei.<anonymous> (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:7225)
    at Generator.next (<anonymous>)
    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:375)
    at new Promise (<anonymous>)
    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:152)
    at ei.Et (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:7112)
    at ei.<anonymous> (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:16556)
    at Generator.next (<anonymous>)
    at o (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:208)
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/base/common/console.js:103 [Extension Host] rejected promise not handled within 1 second: Error: socket not writable
log @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/base/common/console.js:103
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/base/common/console.js:103 [Extension Host] stack trace: Error: socket not writable    at write.Promise (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugConnection.js:1:1579)    at new Promise (<anonymous>)    at Mt.write (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugConnection.js:1:1515)    at Mt.<anonymous> (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:8544)    at Generator.next (<anonymous>)    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:375)    at new Promise (<anonymous>)    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:152)    at Mt.F (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:8261)    at Mt.G (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:8229)    at B.Promise (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:8001)    at new Promise (<anonymous>)    at Mt.B (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:7976)    at Mt.<anonymous> (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:10359)    at Generator.next (<anonymous>)    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:375)    at new Promise (<anonymous>)    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:152)    at Mt.sendStopCommand (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:10301)    at ei.<anonymous> (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:7225)    at Generator.next (<anonymous>)    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:375)    at new Promise (<anonymous>)    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:152)    at ei.Et (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:7112)    at ei.<anonymous> (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:16556)    at Generator.next (<anonymous>)    at o (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:208)
log @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/base/common/console.js:103
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/platform/log/common/log.js:139   ERR socket not writable: Error: socket not writable
    at write.Promise (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugConnection.js:1:1579)
    at new Promise (<anonymous>)
    at Mt.write (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugConnection.js:1:1515)
    at Mt.<anonymous> (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:8544)
    at Generator.next (<anonymous>)
    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:375)
    at new Promise (<anonymous>)
    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:152)
    at Mt.F (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:8261)
    at Mt.G (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:8229)
    at B.Promise (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:8001)
    at new Promise (<anonymous>)
    at Mt.B (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:7976)
    at Mt.<anonymous> (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:10359)
    at Generator.next (<anonymous>)
    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:375)
    at new Promise (<anonymous>)
    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:152)
    at Mt.sendStopCommand (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/connection.js:1:10301)
    at ei.<anonymous> (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:7225)
    at Generator.next (<anonymous>)
    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:375)
    at new Promise (<anonymous>)
    at e (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:152)
    at ei.Et (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:7112)
    at ei.<anonymous> (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:16556)
    at Generator.next (<anonymous>)
    at o (/Users/lover/.vscode/extensions/devsense.phptools-vscode-1.0.3774/out/src/debug/debugger.js:1:208)
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/browser/parts/notifications/notificationsAlerts.js:29 Unable to open 'debug-eval': Unable to read file (Error: File not found (/Users/lover/Websites/web2/xdebug:/debug-eval)).
onDidNotificationChange @ /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/browser/parts/notifications/notificationsAlerts.js:29
```

**Comment by jakubmisek** [comment]
thank you for the issue, is there any context or steps to take through?

**Comment by tiansin** [comment]
@jakubmisek In the state of being debugged, press F10 to perform the single-step code action, which appears when F10 is pressed once, but only once.

**Comment by jakubmisek** [comment]
Doesn't crash for me, do you have any more details? Maybe a gif would help. Is it a new issue that appeared after the todays update?

**Comment by tiansin** [comment]
@jakubmisek It’s just that I debugged the breakpoint, it happened once in the single step, and it didn’t appear in the back. I can’t take an animation.

**Comment by jakubmisek** [comment]
ok, thank you, I'll forward it to someone who might now, cc @Miloslav : )

**Comment by Miloslav** [comment]
Hi guys,

I was thinking if this was caused by php sending to vscode information about stack frame and instead of file, there would be "debug-eval". 

That can happen e.g. when eval is presented in the callstack and you click on it. So instead of file we would be opening some strange path that contains xdebug:/debug-eval. I've tried that case, but we handled that situation correctly and created a file with name eval.php with the content of eval and opened that one instead. So I'm not sure what to try next. If you'd have more information how to reproduce it would be appreciated.

Thanks
